src/cr/cube/matrix/assembler.py

```diff
@@ -64,15 +64,24 @@ def row_display_order(cls, dimensions, second_order_measures, format):
         HelperCls = (
             _SortRowsByBaseColumnHelper
             if collation_method == CM.OPPOSING_ELEMENT
-            else _SortRowsByDerivedColumnHelper
-            if collation_method == CM.OPPOSING_INSERTION and dim_type in DT.ARRAY_TYPES
-            else _SortRowsByInsertedColumnHelper
-            if collation_method == CM.OPPOSING_INSERTION
-            else _SortRowsByLabelHelper
-            if collation_method == CM.LABEL
-            else _SortRowsByMarginalHelper
-            if collation_method == CM.MARGINAL
-            else _RowOrderHelper
+            else (
```

Comment:
IMHO this is quite difficult to read. Could you think about something more readable?

Reply:
yeah this is crap... I dunno man... will try to figure out something better... maybe the switch?

Reply:
Is this better:

```
    @classmethod
    def factory(cls, cube, dimensions, slice_idx):
        """Return _BaseCubeMeans subclass instance appropriate to `cube`.

        Raises `ValueError` if the cube-result does not include a cube-means measure.
        """
        if cube.means is None:
            raise ValueError("cube-result does not contain cube-means measure")

        def get_cube_means_cls(dim_types):
            if dim_types == (DT.MR, DT.MR):
                return _MrXMrCubeMeans
            if dim_types[0] == DT.MR:
                return _MrXCatCubeMeans
            if dim_types[1] == DT.MR:
                return _CatXMrCubeMeans
            return _CatXCatCubeMeans

        dimension_types = cube.dimension_types[-2:]
        CubeMeansCls = get_cube_means_cls(dimension_types)

        return CubeMeansCls(
            dimensions, cube.means[cls._slice_idx_expr(cube, slice_idx)]
        )
```

Reply:
yes I like it more